OCPBUGS#44579: prevent unintended unsupported sys config (#87660)
openshift/openshift-docs: modules/microshift-updates-rhde-config-rhel-repos.adoc

// Module included in the following assemblies:
//
//microshift_updating/microshift-update-options.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-updates-rhde-config-rhel-repos_{context}"]
= Keeping {microshift-short} and {op-system-base} in a supported configuration

When using RPM updates, avoid creating an unsupported configuration or breaking your cluster by carefully managing your {op-system-base} repositories.

.Repository management steps

* Avoid unintended updates by locking your operating system version with the following command:
+
[source,terminal]
----
$ sudo subscription-manager release --set=x.y
----

* To update both {microshift-short} and {op-system-base} versions, use the following command:
+
[source,terminal]
----

> $ sudo subscription-manager release --set=x.y+2

Reviewer: This might be an odd way to phrase this. A user would not set this to 9.6 since it i not currently available. Any thoughts @ggiguash  ?

Reviewer 2: Not sure how to better frame this, but to me, this makes sense.
Consider e.g. you are on 4.14 on 9.2. If you want to upgrade to 4.16 on 9.4, you have change your "--set=9.2" to "--set=9.4, that is "y+2", BEFORE doing the big "dnf update". And dont forget to also change the OCP repo from 4.14 to 4.16! Otherwise you would end up with 4.14 on 9.6, which would again be an unsupported config.

// Module included in the following assemblies:
//
//microshift_updating/microshift-update-options.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-updates-rhde-config-rhel-repos_{context}"]
= Keeping {microshift-short} and {op-system-base} in a supported configuration

When using RPM updates, avoid creating an unsupported configuration or breaking your cluster by carefully managing your {op-system-base} repositories.

.Repository management steps

> * Avoid unintended updates by locking your operating system version with the following command:

Reviewer 3: Maybe get the "by using the following command" wording throughout the procedure:
```suggestion
* Lock your operating system version to avoid unintended updates by using the following command:
```

+
[source,terminal]
----
$ sudo subscription-manager release --set=x.y
----

> * To update both {microshift-short} and {op-system-base} versions, use the following command:

Reviewer 3: ```suggestion
* Update both {microshift-short} and {op-system-base} versions by running the following command:
```

+
[source,terminal]
----
$ sudo subscription-manager release --set=<9.4> command. # <1>
----
<1> Replace _<9.4>_ with the major and minor version of your compatible {op-system-base} system.

* If you are using an EUS {microshift-short} release, first disable the {op-system-base} standard-support-scope repositories by using the following command:
+
[source,terminal]
----
$ sudo subscription-manager repos \
    --disable=rhel-<9>-for-x86_64-appstream-rpms \ # <1>
    --disable=rhel-<9>-for-x86_64-baseos-rpms
----
<1> Replace _<9>_ with the major version of your compatible {op-system-base} system.
+

> * After you disabled the standard-support repositories, enable the {op-system-base} EUS repos using the following command:

Reviewer 3: ```suggestion
* After you disable the standard-support repositories, enable the {op-system-base} EUS repos by using the following command:
```

Reviewer 3: Should this be a sub-bullet to the previous bullet?

+
[source,terminal]
----
$ sudo subscription-manager repos \
    --enable rhel-<9>-for-x86_64-appstream-eus-rpms \  # <1>
    --enable rhel-<9>-for-x86_64-baseos-eus-rpms`
----
<1> Replace _<9>_ with the major version of your compatible {op-system-base} system.

> .  List the repositories you have enabled for {op-system-base} by running the following command:

Reviewer 3: Consistency throughout the procedure:
```suggestion
.  List the repositories you have enabled for {op-system-base} by using the following command:
```

> .  List the repositories you have enabled for {op-system-base} by running the following command:

Reviewer 3: This bit isn't rendering correctly in the [preview](https://87660--ocpdocs-pr.netlify.app/microshift/latest/microshift_updating/microshift-update-options.html#microshift-updates-rhde-config-rhel-repos_microshift-update-options).
